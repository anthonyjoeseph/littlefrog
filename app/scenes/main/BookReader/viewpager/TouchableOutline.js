// app/BookReader/BookPageImage.js

import React, { Component } from 'react';
import {
  View,
  Image,
  TouchableWithoutFeedback
} from 'react-native';

class TouchableOutline extends Component{
  constructor(props){
    super(props);

    this.state = {
      highlighted: false
    };
  }

  render(){
      return (
        <View
          style={
            {
              position:'absolute',
              left: this.props.left,
              bottom: this.props.bottom,
              width: this.props.width,
              height: this.props.height
            }
          }>
          <TouchableWithoutFeedback
            onPressIn={
              function(){
                this.setState({highlighted: true});
              }.bind(this)}
            onPressOut={
              function(){
                this.setState({highlighted: false});
              }.bind(this)}
            onPress={
              function(){
                this.props.onPress();
              }.bind(this)}
          >
            <Image
              resizeMode='stretch'
              source={{uri: this.props.source}}
              style={{
                tintColor: this.state.highlighted ? this.props.tintColor : null,
                width: this.props.width,
                height: this.props.height
              }} />
          </TouchableWithoutFeedback>
        </View>
      );
  }
}

export default TouchableOutline;
