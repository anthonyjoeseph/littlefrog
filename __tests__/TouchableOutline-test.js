// __tests__/BookInfoPage-test.js
import 'react-native';
import React from 'react';
import TouchableOutline from '../app/scenes/main/BookReader/viewpager/TouchableOutline';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <TouchableOutline
      left={0.5}
      bottom={0.5}
      width={0.15}
      height={0.35}
      tintColor={'red'}
      source={'frog_outline'}
      onPress={ ()=>{} }
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
