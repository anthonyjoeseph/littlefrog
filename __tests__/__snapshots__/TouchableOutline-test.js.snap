exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0.5,
      "height": 0.35,
      "left": 0.5,
      "position": "absolute",
      "width": 0.15,
    }
  }>
  <Image
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    resizeMode="stretch"
    source={
      Object {
        "uri": "frog_outline",
      }
    }
    style={
      Object {
        "height": 0.35,
        "tintColor": null,
        "width": 0.15,
      }
    }
    testID={undefined} />
</View>
`;
